lib/wanda/execution.ex

```diff
@@ -27,21 +27,43 @@ defmodule Wanda.Execution do
   defp maybe_start_execution(_, _, _, [], _, _), do: {:error, :no_checks_selected}
 
   defp maybe_start_execution(execution_id, group_id, targets, checks, env, config) do
-    case DynamicSupervisor.start_child(
-           Supervisor,
-           {Server,
-            execution_id: execution_id,
-            group_id: group_id,
-            targets: targets,
-            checks: checks,
-            env: env,
-            config: config}
-         ) do
-      {:ok, _} ->
-        :ok
+    with false <- execution_for_group_id_already_running(group_id),
+         {:ok, _} <-
+           DynamicSupervisor.start_child(
```

Comment:
Hey @arbulu89 good try, but this is actually not needed. 
By simply changing the execution name to the group_id in the execution server `via_tuple` function, the dynamic supervisor will automatically not start an existing child returning `{:error, {:already_started, pid}`.
So by matching that error in the old code case you could return :group_already_running or whatever we want.

https://github.com/trento-project/wanda/blob/b3dde2dd2da96c33ecd0d514aaa88a8b62b53252/lib/wanda/execution/server.ex#L40

you'll need to change `receive_facts` as it will take both group_id and execution_id and in the handler of the `receive_facts` you could discard execution_ids that are different from the running genserver.

Reply:
@fabriziosestito I thought of that beforehand and I don't really like it. The execution id is the unique identifier of the execution, and we use it to run other functions, like getting the facts from this specific id. If I replace it by the group_id, we could have 2 executions with the same executions ids. And if I combine them, we could still have strange scenarios like accepting the same execution_id with a different group_id as valid. Strange things.

I personally wouldn't go that path (and it requires as well, to be passing the group_id all around the execution messages, etc)

Reply:
> we could have 2 executions with the same executions ids
we know that executions_id are uuid, so they are unique. when would 2 executions with the same execution_id happen?